src/constraints/GenericConstraint.jl

```diff
@@ -41,6 +41,20 @@ end
 
 vexity(c::GenericConstraint) = vexity(vexity(c.child), c.set)
 
+function vexity(
+    vex::Vexity,
+    ::Union{
+        MOI.PositiveSemidefiniteConeSquare,
+        MOI.SecondOrderCone,
+        MOI.ExponentialCone,
```

Comment:
I would much prefer that we kept separate methods for this, even if they are duplicated.